common/authorization/default_jwt_claim_mapper.go

```diff
@@ -130,12 +130,7 @@ func parseJWT(tokenString string, keyProvider TokenKeyProvider) (jwt.MapClaims,
 
 func parseJWTWithAudience(tokenString string, keyProvider TokenKeyProvider, audience string) (jwt.MapClaims, error) {
 
-	var parser *jwt.Parser
-	if strings.TrimSpace(audience) == "" {
-		parser = jwt.NewParser(jwt.WithoutAudienceValidation())
-	} else {
-		parser = jwt.NewParser(jwt.WithAudience(audience))
-	}
+	parser := jwt.NewParser(jwt.WithValidMethods(keyProvider.SupportedMethods()))
 	token, err := parser.Parse(tokenString, func(token *jwt.Token) (interface{}, error) {
 
 		kid, ok := token.Header["kid"].(string)
```

Comment:
is this switch statement here for HmacKey redundant now that we are using the SupportedMethods feature?

Reply:
I don't this it's redundant because we need to call `keyProvider.HmacKey()` in case `keyProvider` (configured as a separate plugin) supports HMAC. We just don't support it in `defaultTokenKeyProvider`. They are indeed currently coupled [here](https://github.com/temporalio/temporal/blob/3c427e6a2d5e7bcdda5868f379a26f6eb578871f/common/authorization/claim_mapper.go#L79), but only for convenience.